algorithms/map-anchoring/src/map-anchoring.cc

```diff
@@ -58,23 +95,37 @@ bool anchorAllMissions(
   std::unordered_map<vi_map::MissionId, int> remaining_trials_for_mission_map;
   vi_map::MissionIdList all_missions;
   map->getAllMissionIds(&all_missions);
+  std::stringstream ss;
+  ss << "\n";
+  ss << "----------------------------------------------------------------\n";
+  ss << "                         MapAnchoring                           \n";
+  ss << "----------------------------------------------------------------\n";
+  ss << " Current State:\n";
   for (const vi_map::MissionId& mission_id : all_missions) {
     const vi_map::Mission& mission = map->getMission(mission_id);
     const vi_map::MissionBaseFrame& base_frame =
         map->getMissionBaseFrame(mission.getBaseFrameId());
-    if (!base_frame.is_T_G_M_known()) {
+    const bool baseframe_is_known = base_frame.is_T_G_M_known();
+    if (!baseframe_is_known) {
       missions_to_anchor.push(mission_id);
       remaining_trials_for_mission_map[mission_id] = kTrialsPerMission;
     }
+    ss << "\t- " << mission_id << ":\t"
+       << ((baseframe_is_known) ? "anchored\n" : "not anchored\n");
   }
-
+  ss << "----------------------------------------------------------------\n";
+  ss << " Result:\n";
   if (missions_to_anchor.size() == all_missions.size()) {
-    LOG(ERROR) << "At least one mission needs to have a known baseframe.";
+    ss << "\tFailed: At least one mission needs to have a known baseframe. \n";
+    ss << "----------------------------------------------------------------\n";
+    LOG(INFO) << ss.str();
     return false;
   }
 
   if (missions_to_anchor.empty()) {
-    LOG(INFO) << "All baseframes are known -- nothing to do.";
+    ss << "\tAll baseframes are known -- nothing to do.\n";
```

Comment:
since the missions have never been selected (happens in the while loop, we don't need to.) I think an early exit is needed here because this way we can write a more informative log output. Otherwise, we would need to catch that special case below.